### DISCUSSION TITLE
Multiline inline functions run order seems counter intuitive

### DISCUSSION THREAD

**Discussion by Magnetuz** [uiua-lang/uiua#253] [Language Development]
## Description
According to the documentation, the multiline inline functions runs top-to-bottom. At a first glance it seems fine, as it is the usual form to read the code. However, if one just add a line break on its single inline function, it will modify the way the function works and break it. 

In another words, adding line breaks to improve readability will break the inline function. I have never seen such a behaviour and sounds very counter intuitive.

This can be verified on the tutorial example [here](https://uiua.org/pad?src=0_2_0__WCDihpAgKAogIOKKnj0u4oehCiAg4oal4oeMLgopClggNQo=).

## Expected behaviour
Adding line breaks to an inline function does not change the way it works.

## Current behaviour
Adding line breaks to an inline function changes the way it works and breaks it, requiring reworking it. More line breaks, more rework.

**Comment by kaikalii** [comment]
I am aware of this behavior.
I usually just add a newline then press alt+Up to move it up.
I don't think it's worth it to make functions run bottom to top.

**Reply by Magnetuz** [reply]
Hi @kaikalii 

Thank you for your fast reply 

Would you mind to explain why it is not worth it? 
Why did you decided on this current behaviour?
Do you think this behaviour could be a highly probable source of bugs or could impact code readability?

Thank you

**Reply by kaikalii** [reply]
I don't think it hurts readability or is likely to be the cause of bugs.
I don't think you should have to switch from reading top-to-bottom to reading bottom-to-top inside a function.
Reading upward would be really weird as well.
